distributed/diagnostics/tests/test_nvml.py

```diff
@@ -9,6 +9,9 @@
 
 
 def test_one_time():
+    if nvml.device_get_count() < 1:
+        pytest.skip("No GPUs available")
```

Comment:
Do we need to do this check on a per-function basis, or could we make the outcome of `nvml.device_get_count() < 1` a variable to be referred to by `@pytest.mark.skipif(...)`? Mostly asking to better understand how nvml initialization works here.

Reply:
I think we could have use `@pytest.mark.skipif` as well, but given all these cumbersome states I've found while trying to get this PR right, I wouldn't be surprised if that finds yet another weird one. With that said, I'll leave it as is for now, but we could try that on a future PR.